src/engine/specification.md

```diff
@@ -197,32 +255,34 @@ This structure contains the attributes required to initiate a payload build proc
 
-6. Client software **MUST** return `-32002: Invalid terminal block` error if a block referenced by `forkchoiceState.headBlockHash` is a PoW block that doesn't satisfy terminal block conditions according to [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#definitions).
+1. Client software **MUST** begin a payload build process building on top of `forkchoiceState.headBlockHash` and identified via `buildProcessId` value if `payloadAttributes` is not `null` and the forkchoice state has been updated successfully. The build process is specified in the [Payload building](#payload-building) section.
 
-7. If any of the above fails due to errors unrelated to the client software's normal `SYNCING` status, the client software **MUST** return an error.
+1. Client software **MUST** respond to this method call in the following way:
+  * `{payloadStatus: {status: SYNCING, latestValidHash: null, validationError: null}, payloadId: null}` if `forkchoiceState.headBlockHash` references an unknown payload or a payload that can't be validated because requisite data is missing
+  * `{payloadStatus: {status: INVALID, latestValidHash: null, validationError: errorMessage | null}, payloadId: null}` obtained from the [Payload validation](#payload-validation) process if the payload is deemed `INVALID`
+  * `{payloadStatus: {status: VALID, latestValidHash: forkchoiceState.headBlockHash, validationError: null}, payloadId: null}` if the payload is deemed `VALID` and the build process hasn't been started
+  * `{payloadStatus: {status: VALID, latestValidHash: forkchoiceState.headBlockHash, validationError: null}, payloadId: buildProcessId}` if the payload is deemed `VALID` and the build process has begun.
 
-##### Payload build process
-The payload build process is specified as follows:
-* Client software **MUST** set the payload field values according to the set of parameters passed into this method with exception of the `suggestedFeeRecipient`. The built `ExecutionPayload` **MAY** deviate the `feeRecipient` field value from what is specified by the `suggestedFeeRecipient` parameter.
-* Client software **SHOULD** build the initial version of the payload which has an empty transaction set.
-* Client software **SHOULD** start the process of updating the payload. The strategy of this process is implementation dependent. The default strategy is to keep the transaction set up-to-date with the state of local mempool.
-* Client software **SHOULD** stop the updating process when either a call to `engine_getPayload` with the build process's `payloadId` is made or [`SECONDS_PER_SLOT`](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#time-parameters-1) (12s in the Mainnet configuration) have passed since the point in time identified by the `timestamp` parameter.
+1. If any of the above fails due to errors unrelated to the normal processing flow of the method, the client software **MUST** return an error.
```

Comment:
So errors in the payload assembling process are returned as JSON_RPC errors directly?

Reply:
Not sure I understand your question correctly. But if an exception occurred during initiating a payload build process then the call of this method must be responded with error.

Generally, this point states that in the case when e.g. a payload is `INVALID` according to EE rules an `INVALID` status must be in the response, but if exception happened during the validation then the call must be responded with error, not an `INVALID` status.

Reply:
What should EL return if we receive the wrong timestamp in payload attributes? Because I think `INVALID` is connected to headBlockHash, not payload attributes. By the wrong timestamp, I mean `headBlockHash.Timestamp >= payloadAttributes.Timestamp`

Reply:
It would mean that CL is faulty, EL may respond with error in this case. There is `-32602: Invalid params` error that fits well into this case